pkg/provider/provider_acc_test.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+var testAccProvider *schema.Provider
+
+var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
+	"snowflake": func() (tfprotov6.ProviderServer, error) {
+		return tf5to6server.UpgradeServer(
+			context.Background(),
+			testAccProvider.GRPCProvider,
+		)
+	},
+}
+
+func init() {
+	testAccProvider = Provider()
+}
+
+func TestAcc_ProviderUsernameAndPasswordAuth(t *testing.T) {
+	user := os.Getenv("SNOWFLAKE_USER")
+	if user == "" {
+		t.Skip("SNOWFLAKE_USER must be set")
```

Comment:
Skip means this test will be treated as a success (currently). Should it be skipped or failed in such a case?